src/controllers/RendererController.php

```diff
@@ -40,6 +45,11 @@ public function __construct($params)
             $this->model = new ErrorModel(__DIR__ . "/../templates/error.php", $error->getMessage());
             // create error rendering view
             $this->view = new ErrorView($this->model);
+
+            // set status code
+            if ($error instanceof IStatusException) {
+                $this->statusCode = $error->getStatus();
+            }
```

Comment:
This could default to `500` if there is an unexpected error

```suggestion
            $this->statusCode = $error instanceof IStatusException ? $error->getStatus() : ResponseEnum::HTTP_INTERNAL_SERVER_ERROR;
```